scipy/stats/stats.py

```diff
@@ -944,8 +944,17 @@ def moment(a, moment=1, axis=0):
             # the input was 1D, so return a scalar instead of a rank-0 array
             return np.float64(0.0)
     else:
-        mn = np.expand_dims(np.mean(a, axis), axis)
-        s = np.power((a - mn), moment)
+        # Subtract mean along the axis, compute element-wise square
+        a_zero_mean = a - np.expand_dims(np.mean(a, axis), axis)
+        a_zero_mean_2 = a_zero_mean**2
+        
+        s = a_zero_mean_2.copy()
+        for k in range(1, mom // 2):
```

Comment:
Yes, if you push more commits to the same branch, it'll update the PR.

Not sure about the mstats test --- the use of `assert_equal` does not look quite right. But if  `mstats.moment` uses an "old" algorithm, it should probably be updated to be in sync with the non-masked version.